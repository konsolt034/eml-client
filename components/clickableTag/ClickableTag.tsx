"use client";

import styled from "styled-components";

interface TagProps {
  onClick: () => void;
}

const Tag = styled.button<TagProps>`
  font-size: 14px;
  color: #fff;
  background-color: #202020;
  border: solid 1px #6c6c6c;
  border-radius: 4px;
  padding: 2px 8px;
  white-space: nowrap;

  transition: all;
  transition-duration: 0.5s;
  transition-delay: 0.1s;

  &:hover {
    border-color: #ffbf00;
    color: #ffbf00;
    cursor: pointer;
  }
`;

export default Tag;
