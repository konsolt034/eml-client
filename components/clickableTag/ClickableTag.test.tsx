import { fireEvent, render } from "@testing-library/react";

import ClickableTag from ".";

describe("Component - ClickableTag", () => {
  let handleClick: jest.Mock;

  beforeEach(() => {
    handleClick = jest.fn();
  });

  it("should render component", () => {
    const { container } = render(
      <ClickableTag onClick={handleClick}>Action</ClickableTag>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render button component", () => {
    const { getByRole } = render(
      <ClickableTag onClick={handleClick}>Adventure</ClickableTag>
    );
    expect(getByRole("button")).toBeTruthy;
  });

  it("should fire the onClick function when clicked", () => {
    const { getByRole } = render(
      <ClickableTag onClick={handleClick}>{"Click Me!"}</ClickableTag>
    );

    fireEvent.click(getByRole("button"));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
