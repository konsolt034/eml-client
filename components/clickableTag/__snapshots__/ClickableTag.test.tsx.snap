// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - ClickableTag should render component 1`] = `
<div>
  <button
    class="sc-bcPKhP TrbCn"
  >
    Action
  </button>
</div>
`;
